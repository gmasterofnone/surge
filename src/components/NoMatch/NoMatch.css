.error-page {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
  background-color: blue;
}

.error-container {
  width: 60%;
  height: 50%;
  font-family: Arial, Helvetica, sans-serif;
  color: white;
}

.whoops {
  width: 100%;
}

.error-redirect {
  color: #41ead4;
}

.error-header, .error-message {
  margin-left: 6%;
}