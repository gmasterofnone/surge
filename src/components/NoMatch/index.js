import React from 'react';
import './NoMatch.css'
import { Link } from 'react-router-dom';

import whoops from '../../assets/whoops.svg'

const NoMatch = () => {
  return(
    <div className='error-page'>
      <div className='error-container'>
        <img className='whoops' src={whoops} alt='404 page'/>
        <h1 className='error-header'>Blue screen....You broke it...</h1>
        <h3 className='error-message'>ERROR 404: CLICK <Link className='error-redirect' to='/'>HERE</Link> TO UN-BREAK</h3>
      </div>
    </div>
  )
}

export default NoMatch;