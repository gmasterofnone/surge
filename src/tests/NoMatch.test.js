import React from 'react';
import NoMatch from '../components/NoMatch';
import { shallow } from 'enzyme';


describe('NoMatch', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NoMatch/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})