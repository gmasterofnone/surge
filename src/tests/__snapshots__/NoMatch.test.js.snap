// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch should match the snapshot 1`] = `
<div
  className="error-page"
>
  <div
    className="error-container"
  >
    <img
      alt="404 page"
      className="whoops"
      src="whoops.svg"
    />
    <h1
      className="error-header"
    >
      Blue screen....You broke it...
    </h1>
    <h3
      className="error-message"
    >
      ERROR 404: CLICK 
      <Link
        className="error-redirect"
        replace={false}
        to="/"
      >
        HERE
      </Link>
       TO UN-BREAK
    </h3>
  </div>
</div>
`;
